modules/core/client/components/Map/MapStyleControl.js

```diff
@@ -0,0 +1,80 @@
+// External dependencies
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { DropdownButton, MenuItem, ButtonGroup } from 'react-bootstrap';
+
+// Internal dependencies
+import {
+  getMapboxToken,
+  MAP_STYLE_MAPBOX_STREETS,
+  MAP_STYLE_MAPBOX_SATELLITE,
+  MAP_STYLE_MAPBOX_OUTDOORS,
+  MAP_STYLE_OSM,
+} from './constants';
+import MapIcon from './MapIcon';
+
+export default function MapStyleControl({ mapStyle, setMapstyle }) {
+  const { t } = useTranslation('core');
+  const mapboxToken = getMapboxToken();
+
+  const mapboxStyleNames = {};
+  mapboxStyleNames[MAP_STYLE_MAPBOX_STREETS] = t('Streets');
+  mapboxStyleNames[MAP_STYLE_MAPBOX_SATELLITE] = t('Satellite');
+  mapboxStyleNames[MAP_STYLE_MAPBOX_OUTDOORS] = t('Outdoors');
+
+  // If it's an object, it'll have a name. Otherwise it's Mapbox URL in string presentation.
+  const selectedStyle = mapStyle?.name || mapStyle;
```

Comment:
This was important so that for OSM I wouldn't be comparing to `[object object]`.

OSM is a Json object, while other styles are just strings with `mapbox://blabla`.

We might in future have some other json objects too  so I felt worthwhile be little prepared.

I also like to keep OSM layer around as a backup we can quickly switch on even in production.